tests/contrib/contrib.py

```diff
@@ -0,0 +1,57 @@
+"""
+TestCases that each integration should inherit.
+"""
+
+class PatchTestCase(object):
+    """
+    TestCase for testing the patch logic of an integration.
+    """
+    def setup_method(self, method):
+        pass
+
+    def teardown_method(self, method):
+        """
+        If tests being done require the module to be deleted/unimported a
+        helper is provided in tests.utils.delete_module.
+        """
+        pass
+
+    def test_patch(self):
+        """
+        The integration should test that each class, method or function that
+        is to be patched is in fact done so.
+
+        For example:
+
+        The redis integration patches the following methods:
+        - redis.StrictRedis.execute_command
+        - redis.StrictRedis.pipeline
+        - redis.Redis.pipeline
+        - redis.client.BasePipeline.execute
+        - redis.client.BasePipeline.immediate_execute_command
+
+        an appropriate ``test_patch`` would be::
+
+            ddtrace.contrib.redis.patch()
+            assert isinstance(redis.StrictRedis.execute_command, wrapt.ObjectProxy)
```

Comment:
@brettlangdon would it make sense to include the helpers for detecting patching in this PR?

Reply:
I'd like to add a base `unittest.TestCase` in `tests/__init__.py`.

That one can contain generic assertions/helpers.

Reply:
Right. I was just thinking of whether we would like to include that base test case and some helpers here. It'd be nice to not introduce these changes that will change very soon after when those helpers come along.

I could add the basics of what you had in the flask PR for iswrapped for example.

Reply:
I was going to open a new PR today for those changes, but if you wanted to do it... don't think it matters much.

I was thinking we have two main base test cases (for now).

- `BaseTestCase(unittest.TestCase)`
  - contains all the generic helpers like `assert_is_wrapped` and others that do not depends on an integration or `tracer`.
- `BaseTracerTestCase(BaseTestCase)`
  - has a `setUp` that configures a `self.tracer = get_dummy_trace()`
  - has asserts for spans/tracers
  - I was also thinking
    - A `@property` `spans` to fetch the current spans from the writer
    - A `reset()` function to pop the traces from the writer (would have a `tearDown` that calls `reset()`)
      - The thinking is that we can do `self.spans` in the tests, rather than `spans = self.get_spans()`, etc etc.



Those are just my pre-coffee thoughts so far.